Automattic/jetpack
# ThreatsDataViews: ThreatModal integration

## Description
Integrates `ThreatModal` into `ThreatsDataViews`. Adds an `actionToConfirm` flag so we can pass action/click context from the main component to the modal to render certain content conditionally.

**Note:**
- This PR is dependant on https://github.com/Automattic/jetpack/pull/40845
- This updates the `ScanResultsDataViews` now `CurrentThreatsDataViews`) to account for the new modal subcomponents, but does not remove and existing modal infrastructure from Protect.
- A followup has been created to add support for bulk threats/actions (and remove Protect modal infrastructure): https://github.com/Automattic/jetpack/pull/40692
- I have also drafted a rough proposal for improving the navigation between bulk and individual views: https://github.com/Automattic/jetpack/pull/40720

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1552
- https://github.com/Automattic/jetpack-scan-team/issues/1561

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Review Storybook for `ThreatModal` and `ThreatsDataViews`
* Ensure the modal renders appropriately when the title or any actions buttons are clicked
* Verify that content is either displayed or hidden in the correct conditions
* Start a fresh JN instance with this branched checked out in the Beta Tester
* Load threats in free and paid version and verify that modals handle as expected
* Ensure no regressions are introduced
